devito/arch/compiler.py

```diff
@@ -456,6 +456,14 @@ def __init__(self, *args, **kwargs):
         else:
             self.cflags += ['-march=native']
 
+        # For mpi, mpicc is compiled against amdclang not aompcc, so need the flags back.
+        if kwargs.get('mpi'):
+            self.ldflags += ['-target', 'x86_64-pc-linux-gnu']
```

Comment:
I know this file obscenely uses `+=` instead of `extend` or `append`, but I'm slowly changing them all, so no more `+=` when making new changes would be good